### ISSUE TITLE
Outlook JS API "getAttachmentContentAsync" doesnt return exact EML content

### ISSUE THREAD

**Issue by mkondadg** [OfficeDev/office-js#5047]
Outlook JS API "getAttachmentContentAsync" doesnt return exact EML content in the callback.

**Your Environment**

Windows desktop
Outlook
You have Microsoft Outlook Version 1.2023.1018.300 (Production).
Client Version is 20231027003.07 .
Windows 10 Enterprise 21H2 19044.2006

You have Microsoft Outlook Version 1.2024.1002.100 (Production).
Client Version is 20241004002.11.
WebView2 Version is 129.0.2792.79.
 
## Expected behavior
Outlook JS API "getAttachmentContentAsync" returns exact EML content in the callback.

## Current behavior
Some tabs are replaced with spaces.
Some new lines are removed
Some X-headers are missing

## Steps to reproduce
Attach provided EML file to a mail and Send it.
Check the value in debugger of "value.content"

**( Note: If we rename .eml to .txt and attach the .txt file and click on Send mail, the sendmail attachment callback returns content in base64 format and content matches exactly )**

## Context
We validate MD5 of attached EML file vs content received in callback and do internal processing based on match.

## Useful logs
Actual file to attach (rename .txt to .eml) :
[js-attachfile.eml.txt](https://github.com/user-attachments/files/17639707/js-attachfile.eml.txt)

Received content in 'getAttachmentContentAsync' callback (rename .txt to .eml) :
[js-attachfile-received-in-callback.eml.txt](https://github.com/user-attachments/files/17639708/js-attachfile-received-in-callback.eml.txt)

Here is the difference screenshot between 2 files

![image](https://github.com/user-attachments/assets/342aeb0f-5d06-4e62-af60-5146d0803917)

**Comment by timwan10** [comment]
To confirm the version numbers you are using look to be from "New Outlook Win32" and Not 
"Classic Outlook Win32"? (Classic Outlook's version # is more like 16.0.XXXXX.YYYYY)

Once you confirm, I can forward this to the "new outlook team". 

But in general, I suspect this won't work. 

To confirm: you are taking an EML file loading it in Outlook. Sending it, and then calling the API on the received mail? (or the mail in the sent items folder?) It would be expected that the client will make some changes to the eml mail during the process.  

In another example:  in "classic Outlook Win32" client it has its own client side MAPI to Mime Converter to produce EML file outputs. It can't be used to produce exact matches.

**Comment by mkondadg** [comment]
Yes, it is "New Outlook Win" but x64

We are attaching an EML file in Outlook to new email and sending it. 
It is NOT from sent items.

But this functionality works fine on Mac New Outlook.

**Comment by timwan10** [comment]
Thanks. I see what you're saying now. That for some reason the .eml file extension name is causing the attachment to be processed differently. I will contact the new Outlook owners about this post.

**Comment by mobisw-msft** [comment]
Hey @mkondadg ,

Sorry if I am a little confused by this. The files you've attached, [js-attachfile.eml.txt](https://github.com/user-attachments/files/17639707/js-attachfile.eml.txt) and [js-attachfile-received-in-callback.eml.txt](https://github.com/user-attachments/files/17639708/js-attachfile-received-in-callback.eml.txt) is the exact same file, and running a text comparison between them doesn't show any difference:
 
<img width="575" alt="image" src="https://github.com/user-attachments/assets/20a93b36-7c9c-4631-85f9-8b2eaccfab9a">

Did you mean to attach a file that has the headers you say are missing? I tried to add them, but unfortunately I can't decipher the header name in the image uploaded.

**Comment by mkondadg** [comment]
@mobisw-msft Oh, I think that was a mistake.

Please check below eml and content returned by outlook.
[22.eml.txt](https://github.com/user-attachments/files/17874429/22.eml.txt)
[22.eml.req.txt](https://github.com/user-attachments/files/17874431/22.eml.req.txt)

Or
Could you reproduce this by saving an email to .eml file and test it ?

**Comment by arpitsangwanmsft** [comment]
Hey @mkondadg!
There is another issue #5068, for getAttachmentContentAsync failing, which is already in our backlog. We can further investigate the eml specific issue once the above-mentioned issue is resolved, will get back once we have an update.
Just to be sure we are on the same page, you are trying to call getAttachmentContentAsync() on the received mail in the receiver's mailbox?

**Comment by mkondadg** [comment]
@arpitsangwanmsft We are calling getAttachmentContentAsync() when user clicks on "Send" button after attaching a file.
We use "OnMessageSend" callback like below in manifest xml
```

              <LaunchEvents>
                <LaunchEvent Type="OnMessageSend" FunctionName="handleSendMail" SendMode="SoftBlock"/>
                <LaunchEvent Type="OnAppointmentSend" FunctionName="handleSendMail" SendMode="SoftBlock"/>
              </LaunchEvents>

```

**Comment by rajjha-msft** [comment]
Hey @mkondadg 

I tried reproducing the same, but I get the same content as in the file.
Is it possible for you to share a test add-in, which we can use to reproduce the issue ?

**Comment by mkondadg** [comment]
@rajjha-msft Is your test on Windows or Mac.
This issue is seen on Windows but works fine on Mac.

Could you post a screenshot of content variable while debugging debugging in your environment and the original file you are testing with ?

**Comment by rajjha-msft** [comment]
Hey @mkondadg 

I was able to reproduce the issue on New Outlook. 
It works fine on OWA, and in New Outlook I am getting incorrect Base64 encoded string, for the file that you have attached. 

Do you have the same observation ?

**Comment by mkondadg** [comment]
Yes, not getting exactly same base64 content of the original file.

**Comment by rajjha-msft** [comment]
Thanks for reporting this issue regarding <enter feature here>. It has been put on our backlog. We unfortunately have no timelines to share at this point 

Internal tracking id: Office: 5062277

Initially we were not able to reproduce the issue because the original file never had `X-MS-Exchange-Organization-RecordReviewCfmType: 0` header. I added it and saw it getting stripped in the response.

**Comment by mmanjaree-msft** [comment]
@mkondadg The fix has been rolled out. Could you please confirm if the issue is resolved for you

**Comment by mkondadg** [comment]
Thanks @mmanjaree-msft . Will test and update you.

**Comment by mmanjaree-msft** [comment]
@mkondadg Please update here in case issue persists
